steward/src/simulate.rs

```diff
@@ -40,47 +31,27 @@ impl steward::simulate_contract_call_server::SimulateContractCall for SimulateHa
             }
         };
         info!("encoded call: {encoded_call}");
-        let response = if request.encode_only {
+        let response_body = if request.encode_only {
             String::default()
         } else {
             info!("simulating call to {cellar_id} with tenderly");
-            tenderly::simulate(cellar_id, encoded_call.clone()).await?
+            let response = tenderly::simulate(cellar_id, encoded_call.clone()).await?;
+            info!("response object: {response:?}");
+            response.text().await.map_err(|e| {
+                Status::new(
+                    Code::Internal,
+                    format!("failed to deserialize response body: {e}"),
+                )
+            })?
         };
 
         Ok(Response::new(SimulateResponse {
             encoded_call,
-            response_body: response,
+            response_body,
         }))
     }
 }
 
 pub fn get_string_encoded_call(request: ScheduleRequest) -> Result<String, Error> {
-    if request.call_data.is_none() {
-        return Err(ErrorKind::Http.context("empty contract call data").into());
-    }
-
-    match request.call_data.unwrap() {
-        AaveV2Stablecoin(call) => {
-            if call.function.is_none() {
-                return Err(ErrorKind::Http.context("empty function data").into());
-            }
-
-            aave_v2_stablecoin::get_encoded_call(call.function.unwrap(), request.cellar_id)
-        }
-        CellarV1(call) => {
-            if call.function.is_none() {
-                return Err(ErrorKind::Http.context("empty function data").into());
-            }
-
-            cellar_v1::get_encoded_call(call.function.unwrap(), request.cellar_id)
-        }
-        CellarV2(call) => {
-            if call.function.is_none() {
-                return Err(ErrorKind::Http.context("empty function data").into());
-            }
-
-            cellar_v2::get_encoded_call(call.function.unwrap(), request.cellar_id)
-        }
-    }
-    .map(|b| bytes_to_hex_str(&b))
+    get_encoded_call(request).map(|b| bytes_to_hex_str(&b))
```

Comment:
`map` is applied to elements of an iterator. Unless `get_encoded_call` is returning an array of calls to encode, isn't this going to be calling the `map` function on every element of the incoming` vec`? I'm confused as to why this would be working.

Reply:
It's opaque but this is a case of the `Result<T, E>` type's `map` function which simply changes the `T` to another type `U`:

https://doc.rust-lang.org/std/result/enum.Result.html#method.map